prems-resources/backend/src/main/java/io/uhndata/cards/prems/internal/importer/AbstractConditionalClarityDataProcessor.java

```diff
@@ -0,0 +1,182 @@
+            ConditionDefinition def = new ConditionDefinition(conditionString);
+            if (def.isValid()) {
+                this.conditions.add(def);
+            }
+        }
+    }
+
+    protected abstract Map<String, String> handleUnmatchedCondition(Map<String, String> input);
+
+    protected abstract Map<String, String> handleAllConditionsMatched(Map<String, String> input);
+
+    @Override
+    public Map<String, String> processEntry(Map<String, String> input)
+    {
+        for (ConditionDefinition condition : this.conditions) {
+            if (!condition.matches(input)) {
+                return this.handleUnmatchedCondition(input);
+            }
+        }
+        // All filters match if data got here
+        return this.handleAllConditionsMatched(input);
+    }
+
+    @Override
+    public int getPriority()
+    {
+        return this.priority;
+    }
+
+    private enum Operator
+    {
+        NEQ("<>", false, (input, value) -> input == null || !input.equalsIgnoreCase(value)),
+        LTE("<=", false, (input, value) -> compareDouble(input, value, (a, b) -> a <= b)),
+        LT("<", false, (input, value) -> compareDouble(input, value, (a, b) -> a < b)),
+        GTE(">=", false, (input, value) -> compareDouble(input, value, (a, b) -> a >= b)),
+        GT(">", false, (input, value) -> compareDouble(input, value, (a, b) -> a > b)),
+        EQ("=", false, (input, value) -> input != null && input.equalsIgnoreCase(value)),
+        NOT_MATCHES("not matches", false, (input, value) -> input == null || !input.matches(value)),
+        MATCHES("matches", false, (input, value) -> input != null && input.matches(value)),
+        NOT_IN("not in", false, (input, value) -> input == null
+            && !Arrays.stream(value.split("\\s*;\\s*")).anyMatch(input::equalsIgnoreCase)),
+        IN("in", false, (input, value) -> input != null
+            && Arrays.stream(value.split("\\s*;\\s*")).anyMatch(input::equalsIgnoreCase)),
+        EMPTY("is empty", true, (input, value) -> input == null || input.length() == 0),
+        NOT_EMPTY("is not empty", true, (input, value) -> input != null && input.length() > 0);
+
+        private final String operator;
+
+        private final boolean unary;
+
+        private final BiFunction<String, String, Boolean> evaluator;
+
+        Operator(final String operator, final boolean unary, final BiFunction<String, String, Boolean> evaluator)
+        {
+            this.operator = operator;
+            this.unary = unary;
+            this.evaluator = evaluator;
+        }
+
+        Pair<String, String> parse(String configuration)
+        {
+            if (this.unary) {
+                if (configuration.endsWith(this.operator)) {
+                    return Pair.of(StringUtils.removeEnd(configuration, this.operator).trim(), null);
+                }
+            } else {
+                String[] pieces = configuration.split("\\s+" + this.operator + "\\s+", 2);
```

Comment:
This now requires spaces around all operators. We could instead put spaces only around the operators that need them, e.g. `NEQ("<>"...` but `IN(" in "...`, and change this regexp to:

```suggestion
                String[] pieces = configuration.split("\\s*" + this.operator + "\\s*", 2);
```